packages/firestore/src/core/sync_engine_impl.ts

```diff
@@ -316,21 +316,23 @@ export async function syncEngineListen(
       syncEngineImpl.localStore,
       queryToTarget(query)
     );
-    if (syncEngineImpl.isPrimaryClient) {
-      remoteStoreListen(syncEngineImpl.remoteStore, targetData);
-    }
 
     const status = syncEngineImpl.sharedClientState.addLocalQueryTarget(
       targetData.targetId
     );
     targetId = targetData.targetId;
+
     viewSnapshot = await initializeViewAndComputeSnapshot(
       syncEngineImpl,
       query,
       targetId,
       status === 'current',
       targetData.resumeToken
     );
+
```

Comment:
Do local processing first to get expectedCount, then send Target with listen request.